crates/muvm/src/guest/bridge/common.rs

```diff
@@ -0,0 +1,980 @@
+use anyhow::Result;
+use nix::errno::Errno;
+use nix::libc::{c_int, c_void, off_t, O_RDWR};
+use nix::sys::epoll::{Epoll, EpollCreateFlags, EpollEvent, EpollFlags, EpollTimeout};
+use nix::sys::mman::{mmap, munmap, MapFlags, ProtFlags};
+use nix::sys::socket::{recvmsg, sendmsg, ControlMessage, ControlMessageOwned, MsgFlags, RecvMsg};
+use nix::unistd::read;
+use nix::{cmsg_space, ioctl_readwrite, ioctl_write_ptr};
+use std::cell::RefCell;
+use std::collections::{HashMap, HashSet, VecDeque};
+use std::fs::File;
+use std::io::{IoSlice, IoSliceMut, Read, Write};
+use std::net::{TcpListener, TcpStream};
+use std::num::NonZeroUsize;
+use std::os::fd::{AsFd, AsRawFd, BorrowedFd, FromRawFd, OwnedFd, RawFd};
+use std::os::unix::net::{UnixListener, UnixStream};
+use std::ptr::NonNull;
+use std::rc::{Rc, Weak};
+use std::{env, fs, mem, slice, thread};
+
+pub const PAGE_SIZE: usize = 4096;
+
+const VIRTGPU_CONTEXT_PARAM_CAPSET_ID: u64 = 0x0001;
+const VIRTGPU_CONTEXT_PARAM_NUM_RINGS: u64 = 0x0002;
+const VIRTGPU_CONTEXT_PARAM_POLL_RINGS_MASK: u64 = 0x0003;
+const CAPSET_CROSS_DOMAIN: u64 = 5;
+const CROSS_DOMAIN_CHANNEL_RING: u32 = 1;
+const VIRTGPU_BLOB_MEM_GUEST: u32 = 0x0001;
+const VIRTGPU_BLOB_MEM_HOST3D: u32 = 0x0002;
+const VIRTGPU_BLOB_FLAG_USE_MAPPABLE: u32 = 0x0001;
+const VIRTGPU_BLOB_FLAG_USE_SHAREABLE: u32 = 0x0002;
+const VIRTGPU_EVENT_FENCE_SIGNALED: u32 = 0x90000000;
+const CROSS_DOMAIN_ID_TYPE_VIRTGPU_BLOB: u32 = 1;
+
+#[repr(C)]
+#[derive(Default)]
+struct DrmVirtgpuContextInit {
+    num_params: u32,
+    pad: u32,
+    ctx_set_params: u64,
+}
+
+#[repr(C)]
+#[derive(Default)]
+struct DrmVirtgpuContextSetParam {
+    param: u64,
+    value: u64,
+}
+
+#[rustfmt::skip]
```

Comment:
What does `rustfmt` do to these? Just curious... :laughing:

Reply:
Each argument on a separate line

Reply:
Is that *really bad*? :see_no_evil:

Reply:
There is a lot of them and each ends up looking kinda terrible.